pkg/teleporter/relayer.go

```diff
@@ -0,0 +1,377 @@
+)
+
+var teleporterRelayerRequiredBalance = big.NewInt(0).Mul(big.NewInt(1e18), big.NewInt(500)) // 500 AVAX
+
+func GetRelayerKeyInfo(keyPath string) (string, string, error) {
+	var (
+		k   *key.SoftKey
+		err error
+	)
+	if utils.FileExists(keyPath) {
+		ux.Logger.PrintToUser("Loading stored key %q for relayer ops", constants.AWMRelayerKeyName)
+		k, err = key.LoadSoft(models.LocalNetwork.ID, keyPath)
+		if err != nil {
+			return "", "", err
+		}
+	} else {
+		ux.Logger.PrintToUser("Generating stored key %q for relayer ops", constants.AWMRelayerKeyName)
+		k, err = key.NewSoft(0)
+		if err != nil {
+			return "", "", err
+		}
+		if err := k.Save(keyPath); err != nil {
+			return "", "", err
+		}
+	}
+	return k.C(), hex.EncodeToString(k.Raw()), nil
+}
+
+func FundRelayer(
+	rpcURL string,
+	prefundedPrivateKey string,
+	teleporterRelayerAddress string,
+) error {
+	// get teleporter relayer balance
+	client, err := evm.GetClient(rpcURL)
+	if err != nil {
+		return err
+	}
+	teleporterRelayerBalance, err := evm.GetAddressBalance(client, teleporterRelayerAddress)
+	if err != nil {
+		return err
+	}
+	if teleporterRelayerBalance.Cmp(teleporterRelayerRequiredBalance) < 0 {
+		toFund := big.NewInt(0).Sub(teleporterRelayerRequiredBalance, teleporterRelayerBalance)
+		err := evm.FundAddress(
+			client,
+			prefundedPrivateKey,
+			teleporterRelayerAddress,
+			toFund,
+		)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type relayerRunFile struct {
+	Pid int `json:"pid"`
+}
+
+func DeployRelayer(
+	binDir string,
+	configPath string,
+	logFilePath string,
+	runFilePath string,
+	storageDir string,
+) error {
+	if err := RelayerCleanup(runFilePath, storageDir); err != nil {
+		return err
+	}
+	downloader := application.NewDownloader()
+	version, err := downloader.GetLatestReleaseVersion(binutils.GetGithubLatestReleaseURL(constants.AvaLabsOrg, constants.AWMRelayerRepoName))
+	if err != nil {
+		return err
+	}
+	ux.Logger.PrintToUser("using latest awm-relayer version (%s)", version)
+	versionBinDir := filepath.Join(binDir, version)
+	binPath, err := installRelayer(versionBinDir, version)
+	if err != nil {
+		return err
+	}
+	pid, err := executeRelayer(binPath, configPath, logFilePath)
+	if err != nil {
+		return err
+	}
+	return saveRelayerRunFile(runFilePath, pid)
+}
+
+func RelayerCleanup(runFilePath string, storageDir string) error {
+	if err := os.RemoveAll(storageDir); err != nil {
+		return err
+	}
+	if !utils.FileExists(runFilePath) {
+		return nil
+	}
+	bs, err := os.ReadFile(runFilePath)
+	if err != nil {
+		return err
+	}
+	rf := relayerRunFile{}
+	if err := json.Unmarshal(bs, &rf); err != nil {
+		return err
+	}
+	proc, err := os.FindProcess(rf.Pid)
+	if err != nil {
+		// so much expected after a reboot
+		if err := os.Remove(runFilePath); err != nil {
+			return fmt.Errorf("failed removing relayer run file %s: %w", runFilePath, err)
+		}
+		return nil
+	}
+	if err := proc.Signal(syscall.Signal(0)); err != nil {
```

Comment:
why do we send `syscall.Signal(0)`. is it a just a liveness check for the process to see if it exists. 
If so why we rm pid file and return? 
Or maybe I don't understand this code

Reply:
it is a check to see if the process exists. FindProcess is not enough